### PR TITLE
Support capability PTE permissions.

### PR DESCRIPTION
**Pull request by bsdjhb**
- Remove cap permissions from ATTR_DEFAULT.  They should only be
  added in some cases.  This undoes some of the LO/HI changes (and
  the associated upstream commit is now perhaps not needed).  Since
  we can't use a single orr instruction, don't use a PTE_KERN_CAP
  approach as on RISC-V, but instead add a ATTR_CAP_RW used in
  locore for initial PTEs with separate orr instructions under
  #if __has_feature(capabilities).

- Enable ATTR_CAP_RW for the direct map and pmap_qenter().

- Honor VM_PROT_(READ|WRITE)_CAP in pmap_enter(), pmap_enter_quick(),
  and pmap_enter_object().

- Handle LC_SC data aborts.  For the kernel, honor pcb_onfault
  (failing with EFAULT) or panic.  For userland, raise SIGSEGV with
  the appropriate code.  Possibly si_addr for the SIGSEGV should
  be the virtual address from far rather than the PC value in elr.

### CODE REVIEW

**Comment by bsdjhb** [comment]
`cheritest_vm_notag_tmpfile_shared` now passes for both cheritest and cheriabitest.

File: sys/arm64/arm64/trap.c [hunk 1]
```diff
@@ -224,6 +226,38 @@ cap_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
 	userret(td, frame);
 }
 
+static void
+cap_mmu_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
```
Comment (jrtc27) [inline sys/arm64/arm64/trap.c:230]: This is going to be short-lived, right, and be folded into data_abort like on RISC-V?

File: sys/arm64/arm64/trap.c [hunk 2]
```diff
@@ -224,6 +226,38 @@ cap_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
...
+	uintcap_t onfault;
+	struct pcb *pcb;
+
+	pcb = td->td_pcb;
+	if (!lower) {
+		if (td->td_intr_nesting_level == 0 &&
+		    pcb->pcb_onfault != 0) {
+			frame->tf_x[0] = EFAULT;
+			onfault = pcb->pcb_onfault;
+#if __has_feature(capabilities)
```
Comment (jrtc27) [inline sys/arm64/arm64/trap.c:242]: Ehhhh whole function should be :)

File: sys/arm64/arm64/trap.c [hunk 3]
```diff
@@ -110,6 +111,7 @@ static abort_handler *abort_handlers[] = {
 	[ISS_DATA_DFSC_CAP_SEALED] = cap_abort,
 	[ISS_DATA_DFSC_CAP_BOUND] = cap_abort,
 	[ISS_DATA_DFSC_CAP_PERM] = cap_abort,
+	[ISS_DATA_DFSC_LC_SC] = cap_mmu_abort,
```
Comment (jrtc27) [inline sys/arm64/arm64/trap.c:114]: Should probably only be installed for Morello kernels... but same goes for the others

File: sys/arm64/arm64/trap.c (hunk 1 above)
Comment (bsdjhb) [reply sys/arm64/arm64/trap.c:230]: I guess I will look at folding it in now, sure.

File: sys/arm64/arm64/trap.c (hunk 3 above)
Comment (bsdjhb) [reply sys/arm64/arm64/trap.c:114]: Yeah.

File: sys/arm64/arm64/trap.c [hunk 4]
```diff
@@ -91,8 +91,9 @@ typedef void (abort_handler)(struct thread *, struct trapframe *, uint64_t,
     uint64_t, int);
 
 static abort_handler align_abort;
+#if __has_feature(capabilities)
 static abort_handler cap_abort;
-static abort_handler cap_mmu_abort;
+#endif
```
Comment (jrtc27) [inline sys/arm64/arm64/trap.c]: Might be inclined to separate this out into a separate block, although it'll probably stay as just the one so fine as is.

Comment (bsdjhb) [reply sys/arm64/arm64/trap.c]: Yes, I don't expect we will add any more CHERI-specific abort handlers at this point.

**Pull request merged**